### ISSUE TITLE
storage: TestPostPolicyV4 failed

### ISSUE THREAD

**Issue by codyoss** [GoogleCloudPlatform/golang-samples#2162]
See full log [here](https://source.cloud.google.com/results/invocations/8110b3b8-edbf-468a-ba7b-750304ce2736/targets).

Failure message:

```
=== RUN   TestPostPolicyV4
    objects_test.go:488: status code in response mismatch: got 404 want 204
        Body: HTTP/2.0 404 Not Found
        Content-Length: 188
        Content-Type: application/xml; charset=UTF-8
        Date: Thu, 22 Jul 2021 16:08:07 GMT
        Server: UploadServer
        Vary: Origin
        X-Guploader-Uploadid: XXXXXXXXX

        <?xml version='1.0' encoding='UTF-8'?><Error><Code>NoSuchBucket</Code><Message>The specified bucket does not exist.</Message><Details>The specified bucket does not exist.</Details></Error>
    objects_test.go:501: Failed to retrieve attributes: storage: object doesn't exist
--- FAIL: TestPostPolicyV4 (4.24s)
```

**Comment by codyoss** [comment]
This failed on the presubmit for a recent PR I made: #2161. Changes were unrelated so I manually merged that PR and filed this issue. I agree that this seems like a permissions thing.

**Comment by BrennaEpp** [comment]
Thanks for the response @codyoss. An issue for this has been filed internally. As we have been unable to reproduce this error locally, for the moment, this issue will be closed with #2168 which improves logging in case this error happens again. Please let us know if it does.
